spring-cloud-azure-markdown-appdendix-generator/src/main/java/com/azure/spring/cloud/internal/Main.java

```diff
@@ -0,0 +1,217 @@
+			return;
+		}
+		new Generator().generate(outputFile, inclusionPattern, getCurrentDate());
+	}
+
+	static String getCurrentDate() {
+		SimpleDateFormat simpleDateFormat = new SimpleDateFormat();
+		simpleDateFormat.applyPattern("MM/dd/yyyy");
+		Date date = new Date();
+		return simpleDateFormat.format(date);
+	}
+
+	/**
+	 * Generate Markdown files based on prefix and filename.
+	 */
+	static class Generator {
+
+		/**
+		 * Generate files prefixed with _configuration-properties.
+		 * @param outputFile the output file
+		 * @param inclusionPattern the inclusion pattern
+		 */
+		void generate(String outputFile, String inclusionPattern, String date) {
+			try {
+				System.out.println("Parsing all configuration metadata");
+				List<Resource> resources = getSpringConfigurationMetadataJsonFilesInClasspath();
+				Map<String, Object> descriptions = new HashMap<>();
+				int count = 0;
+				int propertyCount = 0;
+				Pattern pattern = Pattern.compile(inclusionPattern);
+				ObjectMapper objectMapper = new ObjectMapper();
+				for (Resource resource : resources) {
+					byte[] bytes = StreamUtils.copyToByteArray(resource.getInputStream());
+					Map<String, Object> rootMap = objectMapper.readValue(bytes, HashMap.class);
+					List<Map<String, Object>> properties = (List<Map<String, Object>>) rootMap.get("properties");
+					count++;
+					propertyCount += properties.size();
+					properties.forEach(propertyItem -> {
+						String name = String.valueOf(propertyItem.get("name"));
+						if (!pattern.matcher(name).matches()) {
+							return;
+						}
+						Object description = propertyItem.get("description");
+						Object defaultValue = propertyItem.get("defaultValue");
+						descriptions.put(name, defaultValue == null ? description
+								: description + " The default value is " + "`" + defaultValue + "`" + ".");
+					});
+				}
+				System.out.println(
+						"Found [" + count + "] Azure projects configuration metadata jsons. [" + descriptions.size()
+								+ "/" + propertyCount + "] were matching the pattern [" + inclusionPattern + "]");
+				System.out.println("Successfully built the description table");
+				if (descriptions.isEmpty()) {
+					System.out.println("Will not update the table, since no configuration properties were found!");
+					return;
+				}
+				generatePropertiesFile(outputFile, descriptions, date);
+			}
+			catch (IOException e) {
+				throw new IllegalStateException(e);
+			}
+		}
+
+		/**
+		 * Get all "spring-configuration-metadata.json" files in classpath.
+		 */
+		protected List<Resource> getSpringConfigurationMetadataJsonFilesInClasspath() throws IOException {
+			Resource[] resources = new PathMatchingResourcePatternResolver()
+					.getResources("classpath*:/META-INF/spring-configuration-metadata.json");
+			return Arrays.stream(resources).filter(resource -> {
+				try {
+					return resource.getURL().toString().contains("azure");
```

Comment:
Did you check this? Does this condition can filter out all target files and exclude all unnecessary files?

Reply:
yes, it can